### ISSUE TITLE
YAML and other meta data - Is this mic on?

### ISSUE THREAD

**Issue by writtenfool** [readwiseio/obsidian-readwise#47]
- I'm on the fence with the Readwise Reader. Much has changed in both Readwise Reader and Obsidian in the last two years.

- I cannot add and status Obsidian yaml fields to readwise notes; they are deleted on the next refresh. 

- And if I add a tag (doc or highlight) in Reader, it does not trigger an update. 

- Given the standardization of yaml (properties) in Obsidian, are you folks going to address this area of this interface?

- I really DO NOT want to maintain metadata in two places. 

- That said, I imagine you folks are up to your ears in keeping up with the flood of new folks into this space and fully realize not too many folks use this integration past the "out of the box" template. 

- My quandary - " I really like Reader! ".

**Comment by Amparose** [comment]
They really need to give us proper, full template instead of the piecemeal segments. Customising where the h1 tag goes, for instance, would fix the metadata/properties issue.

It's frustratingly close to perfect but far enough that I am hesitant to fully integrate it into my system.

**Comment by Amparose** [comment]
After fiddling with this a bit, there are definitely issues that need to be resolved to avoid generating invalid front-matter:

- Multiple authors create line-breaks without proper formatting as a list or just single-line comma separation.
- Titles don't strip colons. 
- {{category}} breaks it.
- {{tags}} need to be formatted as a YAML list.

So yeah, you're right, this plugin isn't great past the "out of the box" template, which is now out of date.

**Comment by writtenfool** [comment]
Regarding YAML, if you create a property in a readwise generated note in Obsidian, the readwise plugin isn't aware of it. So, if you delete the readwise file to refresh it on the next running of the plugin, that yaml element is destroyed. This means you can't do anything with a readwise file on the Obsidian side except refer to it with links in another note. 

In the Readwise and Obsidian discord, several folks provided jinja code that parses multiple authors.

I know it's possible to navigate the obsidian headers programmatically in an Obsidian note, Dataview does this quite well with file.lists .

Cheers!

**Comment by tyler-dot-earth** [comment]
While frustrating for some use-cases, I think these are compelling reasons to [treat readwise exports as read-only](https://notes.tyler.earth/readwise/treat-readwise-exports-as-read-only/) given how the plugin works right now. You may also find value in [my readwise to obsidian template configurations](https://notes.tyler.earth/readwise/readwise-to-obsidian-exports-settings) — I've solved a few of the issues you have talked mentioned.

That said, I share many of these frustrations - particularly the lack of more template variables across all export sections.

**Comment by writtenfool** [comment]
Is it possible for the plugin to check if a incoming note exists in Obsidian, and, if so, preserve whatever is in the yaml (properties) and replace the rest of the note with the incoming content? Just add an option in the plugin for how to handle this conflict.

Or, maybe I don't understand where the work is done in this plugin. Are the markdown files created on the Readwise digs and then simply inserted as new notes in Obsidian?

Thanks, btw, appreciate you folks looking into this interface and adding functionality!

Cheers!

**Comment by tyler-dot-earth** [comment]
@writtenfool 
> Is it possible for the plugin to check if a incoming note exists in Obsidian, and, if so, preserve whatever is in the yaml (properties) and replace the rest of the note with the incoming content? Just add an option in the plugin for how to handle this conflict.
> 
> Or, maybe I don't understand where the work is done in this plugin. Are the markdown files created on the Readwise digs and then simply inserted as new notes in Obsidian?

For the specific pain of not clobbering existing properties, that may be able to be fixed in the plugin.

A quick look into the code and here's what I'm seeing:
- [this code downloads the whole archive, unzips it, and writes the files](https://github.com/readwiseio/obsidian-readwise/blob/4de9c9331e939e3c6ef6289071504592b2cb141a/src/main.ts#L236-L318)
- [these lines merge some existing and new content](https://github.com/readwiseio/obsidian-readwise/blob/4de9c9331e939e3c6ef6289071504592b2cb141a/src/main.ts#L290-L294)
- [this code refreshes a specific book](https://github.com/readwiseio/obsidian-readwise/blob/4de9c9331e939e3c6ef6289071504592b2cb141a/src/main.ts#L355-L386)
- [this code reimports a specific file](https://github.com/readwiseio/obsidian-readwise/blob/4de9c9331e939e3c6ef6289071504592b2cb141a/src/main.ts#L395-L419)

It's also worth noting that none of the code in the links above handle frontmatter specifically. 

So tl;dr yes, I think the plugin could be modified to avoid clobbering the YAML - it would just be additional logic.

I don't work for Readwise, but I will consider working on this if I get some time. Other contributors should feel free to work on it as well.

I'm currently working on improving the reliability of sync functionality - [see this comment if you're curious](https://github.com/readwiseio/obsidian-readwise/issues/54#issuecomment-2156225761).

> Thanks, btw, appreciate you folks looking into this interface and adding functionality!

:heart_hands:

**Comment by writtenfool** [comment]
@tyler-dot-earth - you are a SAINT ! 

We should probably start using properties instead of yaml and frontmatter. Yaml is the markdown name for this top section in a note and frontmatter is the obsidian api name for the same thing, but properties is the feature in Obsidian that folks know to assign meta data to notes.

Bummer about the archive, if only the plugin could iterate it locally, note by note.

Are you a Obsidian user? Reason I bring this up, most folks won't see this discussion. There's only six of us that are watching. If we want more feedback, we might consider mentioning these conversations on the Obsidian Discord.

And, btw,  tl;dr is ok with me; happy to help. 

I'd come back as a subscriber to Reader and Readwise in a heartbeat if some of this stuff could happen in the interface. Reader is freaking awesome.

**Comment by tyler-dot-earth** [comment]
> @tyler-dot-earth - you are a SAINT !

:angel: 

> @writtenfool: We should probably start using properties instead of yaml and frontmatter. Yaml is the markdown name for this top section in a note and frontmatter is the obsidian api name for the same thing, but properties is the feature in Obsidian that folks know to assign meta data to notes.

Note quite. Frontmatter is metadata at the beginning of the file - typically YAML at the beginning of a markdown file (as in Obsidian's case). Properties are Obsidian's branding for the markdown's frontmatter - it's just a friendly interface on top of the YAML. Open any of your notes with properties - they're just YAML frontmatter.

> Bummer about the archive, if only the plugin could iterate it locally, note by note.

It can. That's exactly what's going on here: https://github.com/readwiseio/obsidian-readwise/blob/4de9c9331e939e3c6ef6289071504592b2cb141a/src/main.ts#L265-L268

> Are you a Obsidian user? Reason I bring this up, most folks won't see this discussion. There's only six of us that are watching. If we want more feedback, we might consider mentioning these conversations on the Obsidian Discord.

I am an Obsidian user. The Obsidian discord, Readwise discord, [Readwise subreddit](https://www.reddit.com/r/readwise/) are good places to rally for your cause. You might also consider just submitting a feature request in the app so it can end up on [their feature tracking board](https://readwise.canny.io/reader-features).

> I'd come back as a subscriber to Reader and Readwise in a heartbeat if some of this stuff could happen in the interface. Reader is freaking awesome.

Great info for the Readwise team, though worth noting that I'm just a random contributor :)

**Comment by writtenfool** [comment]
Lemme think on this, I don't have time and you don't either. If I reach out to Obsidian discord or subreddit folks, they would get excited and up for it, but no body available to do the work; letdown ain't healthy.

I am not a current readwise subscriber so their feature tracking isn't available.

My interest is that there is value in keeping this plugin updated routinely and, from my vantive the Readwise folks have dumped us :(  

They were such hard chargers when they announced this on the Obsidian discord three years ago. But not a peep since.

Well, if nothing else, we've left a small kindle here, maybe somebody will notice this Obsidian SOS. 

Cheers!
